http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -118,131 +120,97 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
       )
       createVmJobId = WsmJobId(s"create-vm-${runtime.id.toString.take(10)}")
 
-      runtimeResourcesResult <- createRuntime(
-        CreateAzureRuntimeParams(
+      // send create disk message to WSM
+      createDiskResult <- createDiskForRuntime(
+        CreateAzureDiskParams(msg.workspaceId, runtime, msg.useExistingDisk, azureConfig)
+      )
+
+      // all other resources (hybrid connection, storage container, vm)
+      // are created within the async task
+      _ <- monitorCreateRuntime(
```

Comment:
nit: might be worth looking into `PollRuntimeParams` case class and seeing if it can be simplified at all with this refactoring. For example, `createVmJobId` might only be used in the `monitorRuntimeCreate` function right now, and be extra noise to create here and pass along